scalameter-core/src/main/scala/org/scalameter/Measurer.scala

```diff
@@ -72,22 +96,19 @@ object Measurer {
   object Default {
     def apply() = new Default
 
-    def withNanos() = new Default(conversionFun = _.toDouble) {
-       override def units = "ns"
+    def withNanos() = new Default map { case Quantity(v, _) => 
+      Quantity(v * 1000000.0, "ns") 
     }
   }
 
   /** A default measurer executes the test as many times as specified and returns the sequence of measured times. */
-  class Default private(conversionFun: Long => Double) extends Timer with IterationBasedValue {
-    def this() {
-      this(conversionFun = _ / 1000000.0)
-    }
-
+  class Default extends Timer with IterationBasedValue {
     def name = "Measurer.Default"
 
-    def measure[T, U](context: Context, measurements: Int, setup: T => Any, tear: T => Any, regen: () => T, snippet: T => Any): Seq[Double] = {
+    def measure[T](context: Context, measurements: Int, setup: T => Any,
+      tear: T => Any, regen: () => T, snippet: T => Any): Seq[Quantity[Double]] = {
       var iteration = 0
-      var times = List[Double]()
+      val times = mutable.ListBuffer.empty[Quantity[Double]]
```

Comment:
I've changed it to be more consistent, but don't know if it could break something:
Earlier:
- time-based measurers were prepending result
- memory and gc count were prepending and then doing `reverse` at the end